node/service/src/parachains_db/upgrade.rs

```diff
@@ -49,26 +50,38 @@ impl From<Error> for io::Error {
 }
 
 /// Try upgrading parachain's database to the current version.
-pub fn try_upgrade_db(db_path: &Path) -> Result<(), Error> {
+pub(crate) fn try_upgrade_db(db_path: &Path, db_kind: DatabaseKind) -> Result<(), Error> {
 	let is_empty = db_path.read_dir().map_or(true, |mut d| d.next().is_none());
 	if !is_empty {
-		match current_version(db_path)? {
-			0 => migrate_from_version_0_to_1(db_path)?,
-			CURRENT_VERSION => (),
-			v => return Err(Error::FutureVersion { current: CURRENT_VERSION, got: v }),
+		match get_db_version(db_path)? {
+			// 0 -> 1 migration
+			Some(0) => migrate_from_version_0_to_1(db_path, db_kind)?,
+			// Already at current version, do nothing.
+			Some(CURRENT_VERSION) => (),
+			// This is an arbitrary future version, we don't handle it.
+			Some(v) => return Err(Error::FutureVersion { current: CURRENT_VERSION, got: v }),
+			// No version file. For `RocksDB` we dont need to do anything.
+			None if db_kind == DatabaseKind::RocksDB => (),
```

Comment:
Why don't we need to do anything for rocksdb?

Reply:
If no db version found, it probably means there's no db, so we should just write the current version, which we do below.

Reply:
As it was there before, shouldn't it had also already being created for parity db?

Reply:
The version file was not created previously for ParityDB, seems that this was somehow missed.

Reply:
Previously, we only called `upgrade::try_upgrade_db` for RocksDB, not ParityDB. So ParityDB's parachain db didn't have any versioning. That was an oversight.